src/acquisition/covidcast/csv_importer.py

```diff
@@ -282,17 +279,72 @@ def validate_missing_code(row, attr_quantity, attr_name, filepath=None, logger=N
 
     return missing_entry
 
+  @staticmethod
+  def geo_id_sanity_check(geo_type, geo_id):
```

Comment:
To avoid a lot of `if` statements I changed sanity check `geo_id` with respect to `geo_type`  to the static method that maps geo_types to specific check method, but  receive [tests failures](https://github.com/cmu-delphi/delphi-epidata/actions/runs/4304786994/jobs/7506301536) in similar code 
```python
    def check_hrr():
      if not 1 <= int(geo_id) <= 500:
        return (None, 'geo_id')
      return geo_id
```
and  (next one is passing tests)

```python
    elif geo_type == 'hrr':
      if not 1 <= int(geo_id) <= 500:
        return (None, 'geo_id')
```

possibly I missed something

Reply:
@melange396 @rzats @dmytrotsko could you pls look at issue with fresh mind

Reply:
when `geo_type` is "state" or "nation", it uses a non-numeric `geo_id`.  that causes an exception because `int(geo_id)` is always called, no matter what `geo_type` is (at lines 293, 303, & 314).  see my suggestion below for a possible fix to delay evaluating your `check_xxx()` methods...